cpp/src/CMakeLists.txt

```diff
@@ -255,13 +466,24 @@ file(MAKE_DIRECTORY ${root_directory}/releases)
 add_library(spark_columnar_jni SHARED ${SPARK_COLUMNAR_PLUGIN_SRCS} ${THIRDPARTY_GANDIVA_SRCS})
 add_dependencies(spark_columnar_jni jni_proto)
 
+# Build Arrow
+message(STATUS "Building ARROW from Source: ${BUILD_ARROW}")
+if(BUILD_ARROW)
+  build_arrow(${STATIC_ARROW})
+  message(STATUS "Building Static ARROW: ${STATIC_ARROW}")
+else() #
+  find_arrow()
+  message(STATUS "Use existing ARROW libraries")
+endif()
+
 if(BUILD_PROTOBUF)
-target_link_libraries(spark_columnar_jni
-                      LINK_PUBLIC ${ARROW_LIB} ${PARQUET_LIB} ${GANDIVA_LIB}
-                      LINK_PRIVATE protobuf::libprotobuf)
+  message(STATUS "Building ProtoBuf from Source: ${BUILD_PROTOBUF}")
+  target_link_libraries(spark_columnar_jni
+                        LINK_PRIVATE protobuf::libprotobuf)
```

Comment:
missing protobuf building func?

Reply:
The installation function and target_link_libraries function are separated in two parts.
1. call Protobuf function is located on line 34.
2. link function is located on line 480.
This is the previous way to implement protobuf.
To avoid any ambiguous, let me combine them together in the code and refresh the PR.

Reply:
The travis log shows missing protoc even `build_protobuf=ON`

https://github.com/oap-project/native-sql-engine/pull/63/checks?check_run_id=1769022912
```
-- Building ProtoBuf from Source: ON
-- Configuring done
-- Generating done
-- Build files have been written to: /home/runner/work/native-sql-engine/native-sql-engine/cpp/build
Scanning dependencies of target jni_proto
[  1%] Running PROTO compiler on Exprs.proto
/bin/sh: 1: protobuf_ep-install/bin/protoc: not found
```

Reply:
This is a dependency issue, it happened sometimes in previous code.
The root cause would be "jni_proto" must run after protobuf installed.
Have fixed this issue in 2nd commit.